public/app/user-service-registry/user-service-registry.controller.js

```diff
@@ -21,86 +21,411 @@
+            fd.append( 'body', JSON.stringify(data) );
+            $http({
+                method: "POST",
+                url: '/proxy?url=pz-gateway.stage.geointservices.io/job',
+                data: fd,
+                headers: {
+                    "Content-Type": undefined
+                }
+            }).then(function successCallback(html) {
+                if (html.data.status.indexOf("Success") > -1) {
+                    $scope.serviceId = (JSON.parse(html.data.result.text).resourceId);
+                    $scope.jobStatusResult = html.data;
+                    console.log($scope.serviceId);
+                }
+                else {
+                    getRegisterResult(jobId);
+                }
+            }, function errorCallback(response) {
+                console.log("search.controller fail");
+                toaster.pop('error', "Error", "There was an issue with your request.");
+            });
+        }
+        function getResourceResult(resourceId){
+            var data = {
+                "apiKey": "my-api-key-kidkeid",
+                "jobType": {
+                    "type": "get-resource",
+                    "resourceId": resourceId
+                }
+            };
+            var fd = new FormData();
+            fd.append( 'body', JSON.stringify(data) );
+            $http({
+                method: "POST",
+                url: '/proxy?url=pz-gateway.stage.geointservices.io/job',
+                data: fd,
+                headers: {
+                    "Content-Type": undefined
+                }
+            }).then(function successCallback(html) {
+                $scope.resourceResult = JSON.stringify(html.data);
+            }, function errorCallback(response) {
+                console.log("search.controller fail");
+                toaster.pop('error', "Error", "There was an issue with your request.");
+            });
+        }
+        function getExecuteResult(jobId) {
+            var data = {
+                "apiKey": "my-api-key-kidkeid",
+                "jobType": {
+                    "type": "get",
+                    "jobId": $scope.jobId
+                }
+            };
+
+            var fd = new FormData();
+            fd.append( 'body', JSON.stringify(data) );
+            $http({
+                method: "POST",
+                url: '/proxy?url=pz-gateway.stage.geointservices.io/job',
+                data: fd,
+                headers: {
+                    "Content-Type": undefined
+                }
+            }).then(function successCallback(html) {
+                if (html.data.status.indexOf("Success") > -1) {
+                    $scope.dataId = html.data.result.dataId;
+                    getResourceResult($scope.dataId)
+                    $scope.jobStatusResult = html.data;
+                    console.log($scope.serviceId);
+                }
+                else {
+                    getExecuteResult(jobId);
```

Comment:
if the service doesn't come back fairly quickly, it's going to hammer the service (at about 12 requests per second)... This happened this morning for about 5 or 10 minutes.  We should probably try to set a delay between requests and also a timeout it is doesn't just keep making requests if something went wrong